lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -215,7 +231,9 @@ class DocumentImpl extends NodeImpl {
       this._contentType = this._parsingMode === "xml" ? "application/xml" : "text/html";
     }
 
-    this._URL = whatwgURL.parseURL(privateData.options.url === undefined ? "about:blank" : privateData.options.url);
+    const urlOption = privateData.options.url === undefined ? "about:blank" : privateData.options.url;
+    this._URL = whatwgURL.parseURL(urlOption);
+    this[internalConstants.origin] = new whatwgURL.URL(urlOption).origin;
```

Comment:
Since we now have this -impl.js vs. generated wrapper split, there's no need to use symbols for these. Normal `_origin` properties will work fine and won't be exposed to the user.